activities/tokens/getTokensFromMultifactorTokenActivity.ts

```diff
@@ -47,15 +48,20 @@ export async function getTokensFromMultifactorTokenActivity(
 		)
 	}
 
-	if (
-		!verifyOtpTokenActivity(otpToken, userAccount.otpSecret) &&
-		!(await verifyBackupMultifactorCode(
-			(challengeTokenObject as IChallengeToken).emailAddress,
-			otpToken,
-			context.dataContext
-		))
-	) {
-		throw new GrayskullError(GrayskullErrorCode.InvalidOTP, `Token is incorrect`)
+	const otpSecret = decrypt(userAccount.otpSecret)
+	if (!otpSecret) {
+		throw new GrayskullError(GrayskullErrorCode.NotAuthorized, `Failed to decrypt otp secret`)
+	} else {
+		if (
+			!verifyOtpTokenActivity(otpSecret, otpToken) &&
+			!(await verifyBackupMultifactorCode(
+				(challengeTokenObject as IChallengeToken).emailAddress,
+				otpToken,
+				context.dataContext
+			))
+		) {
+			throw new GrayskullError(GrayskullErrorCode.InvalidOTP, `Token is incorrect`)
+		}
```

Comment:
Just a suggestion, this might read a little better and we can keep it less nested if:

```suggestion
	if (!otpSecret) {
		throw new GrayskullError(GrayskullErrorCode.NotAuthorized, `Failed to decrypt otp secret`)
        }

        if (!verifyOtpTokenActivity(otpSecret, otpToken) &&
			!(await verifyBackupMultifactorCode(
				(challengeTokenObject as IChallengeToken).emailAddress,
				otpToken,
				context.dataContext
			))
		) {
			throw new GrayskullError(GrayskullErrorCode.InvalidOTP, `Token is incorrect`)
	}
```

Throw should exit early so this should prevent the execution of code later on if there is a problem